pkg/client/dtclient/settings_client.go

```diff
@@ -563,24 +563,48 @@ func doObjectsMatchBasedOnUniqueKeys(uniqueKeys []string, source SettingsObject,
 	return true, matches, nil
 }
 
-func isSameValueForKey(key string, c1 []byte, c2 []byte) (same bool, matchingVal any, err error) {
+func isSameValueForKey(targetPath string, c1 []byte, c2 []byte) (same bool, matchingVal any, err error) {
 	u := make(map[string]any)
 	if err := json.Unmarshal(c1, &u); err != nil {
-		return false, nil, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+		return false, nil, fmt.Errorf("failed to unmarshal data for key %q: %w", targetPath, err)
 	}
-	v1 := u[key]
 
+	keys := explodePath(targetPath)
+	value1 := recursiveSearch(u, keys)
+
+	u = nil
 	if err := json.Unmarshal(c2, &u); err != nil {
-		return false, nil, fmt.Errorf("failed to unmarshal data for key %q: %w", key, err)
+		return false, nil, fmt.Errorf("failed to unmarshal data for key %q: %w", targetPath, err)
 	}
-	v2 := u[key]
 
-	if cmp.Equal(v1, v2) {
-		return true, v1, nil
+	value2 := recursiveSearch(u, keys)
+
+	if value1 != nil && value2 != nil && cmp.Equal(value1, value2) {
```

Comment:
Shouldn't it just be `cmp.Equal(v1, v2)` here?
If both keys are `null`, it should return true as well, shouldn't it?

Reply:
This is the case where we want to prevent a nil matching a nil. The empty value of a constrain should not match.

Reply:
True. Could you pls add a comment 🥺, because I won't know this detail in 2 weeks? :D